File: week7/partialabstraction.java
abstract class Shape
{
    abstract float area();
    void display()
    {
        System.out.println("This is a shape");
    }
}

class Rectangle extends Shape
{
    float a,b;
    Rectangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return a*b;
    }
}

class Triangle extends Shape
{
    float a,b;
    Triangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return (float)0.5*a*b;
    }
}
public class PartialAbstraction
{
    public static void main(String[] args) 
    {
        Rectangle obj=new Rectangle(18,7);
        Triangle obj1=new Triangle(17,88);
        System.out.println("Area of Rectangle : "+obj.area());
        System.out.println("Area of Triangle : "+obj1.area());
    }
}

File: week7/partialabstraction.cpp
 #include<iostream>
using namespace std;

class Shapes
{
    private:
    virtual float area()=0;
    public:
    void displayMSG()
    {
        cout<<"This is a shape\n";
    }
};

class Triangle: public Shapes
{
    public:
    float b,h;
    Triangle(float base,float height)
    {
        b=base;
        h=height;
    }
    float area()
    {
        return 0.5*b*h;
    }
};

class Circle: public Shapes
{
    public:
    float r;
    Circle(float radius)
    {
        r=radius;
    }
    float area()
    {
        return 3.14*r*r;
    }
};

int main()
{
    Triangle tri(3,45);
    Circle cir(1.5);
    tri.displayMSG();
    cout<<"Area of the Triangle : "<<tri.area()<<endl;
    cir.displayMSG();
    cout<<"Area of the Circle: "<<cir.area()<<endl;
    return 0;
}
